src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const object = {};
+  const splitDate = date.split(fromFormat[fromFormat.length - 1]);
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    object[fromFormat[i]] = splitDate[i];
+  }
+
+  if (object.hasOwnProperty('YYYY')) {
+    object.YY = object.YYYY.slice(2);
+  }
```

Comment:
The conversion from 'YYYY' to 'YY' should only occur if 'YYYY' is present in the 'fromFormat' and 'YY' is required in the 'toFormat'. You need to check if 'YY' is in the 'toFormat' before slicing the 'YYYY' value.